pkg/platform/kube/functionres/lazy.go

```diff
@@ -2184,23 +2191,68 @@ func (lc *lazyClient) resolveFailFast(pods []v1.Pod,
 				}
 			}
 		}
+	}
 
+	errGroup, _ := errgroup.WithContext(context.TODO(), lc.logger)
+	for _, pod := range pods {
+		pod := pod
 		for _, condition := range pod.Status.Conditions {
 
 			// check if the pod is in pending state, and the reason is that it is unschedulable
 			// (meaning no k8s node can currently run it, because of insufficient resources etc..)
 			if pod.Status.Phase == v1.PodPending && condition.Reason == "Unschedulable" {
 
-				// TODO: if pods has an event that cluster is going to scale up - don't return error
-
-				return errors.Errorf("NuclioFunction pod (%s) is unschedulable", pod.Name)
+				errGroup.Go("WaitAndCheckAutoScaleEvents", func() error {
+					time.Sleep(15 * time.Second)
+
+					triggeredScaleUp, err := lc.isPodAutoScaledUp(pod, namespace)
+					if err != nil {
+						if triggeredScaleUp {
+							return errors.Wrap(err, "Failed to get pod events")
+						}
+						return errors.Wrapf(err,
+							"Autoscaler is unable to schedule NuclioFunction pod (%s)",
+							pod.Name)
+					}
+					return nil
+				})
 			}
 		}
 	}
-
+	if err := errGroup.Wait(); err != nil {
+		return errors.Wrap(err, "NuclioFunction pod is unavailable")
```

Comment:
the error here is not accurate. a more accurate error message is
"Failed to verify at least one pod schedulability" or so.